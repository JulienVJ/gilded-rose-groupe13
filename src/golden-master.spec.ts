import runGoldenMaster from 'jest-golden-master'
import { Item, GildedRose } from '.'

test("Ceci est un test", async () => {
    runGoldenMaster(async () => {
        const item1 = new Item("Aged Brie", 10, 30);
        const itemsArray = [item1];
        const gr = new GildedRose(itemsArray)
        gr.updateQuality()
    })
})

test("Test with quality < 50", async () => {
    runGoldenMaster(async () => {
        const item1 = new Item("Backstage passes to a TAFKAL80ETC concert", 10, 30);
        const itemsArray = [item1];
        const gr = new GildedRose(itemsArray)
        gr.updateQuality()
    })
})

test("Test with quality < 50 but sellIn < 6", async () => {
    runGoldenMaster(async () => {
        const item1 = new Item("Aged Brie", 4, 30);
        const itemsArray = [item1];
        const gr = new GildedRose(itemsArray)
        gr.updateQuality()
    })
})

test("Item is a Sulfuras hammer with a positif quality", async () => {
    runGoldenMaster(async () => {
        const item1 = new Item("Sulfuras, Hand of Ragnaros", 10, 30);

        const itemsArray = [item1];
        const gr = new GildedRose(itemsArray)
        gr.updateQuality()
    })
})

test(
    "Sulfuras hammer with a negatif quality / Random Objet with a positif quality",
    async () => {
        runGoldenMaster(async () => {
            const item1 = new Item("Sulfuras, Hand of Ragnaros", 10, -10);
            const item2 = new Item("Random", 10, 10);
            const itemsArray = [item1, item2];
            const gr = new GildedRose(itemsArray)
            gr.updateQuality()
        })
    })

test(
    "Sulfuras hammer with a positif quality / Random Objet with a negatif quality",
    async () => {
        runGoldenMaster(async () => {
            const item1 = new Item("Sulfuras, Hand of Ragnaros", 10, 10);
            const item2 = new Item("Random", 10, -10);
            const itemsArray = [item1, item2];
            const gr = new GildedRose(itemsArray)
            gr.updateQuality()
        })
    })

test("sellIn inférieur à 0", async () => {
    runGoldenMaster(async () => {
        const item1 = new Item("Aged Brie", -1, 60);
        const itemsArray = [item1];
        const gr = new GildedRose(itemsArray)
        gr.updateQuality()
    })
})

test("sellIn inférieur à 0 et nom != Brie/TAFKAL80ETC/Sulfuras", async () => {
    runGoldenMaster(async () => {
        const item1 = new Item("Different", -1, 0);
        const itemsArray = [item1];
        const gr = new GildedRose(itemsArray)
        gr.updateQuality()
    })
})
test("sellIn inférieur à 0, nom =! sulfuras et quality > 0", async () => {
    runGoldenMaster(async () => {
        const item1 = new Item("Different", -1, 60);
        const itemsArray = [item1];
        const gr = new GildedRose(itemsArray)
        gr.updateQuality()
    })
})

test("sellIn inférieur à 0, nom = TAFKAL80ETC", async () => {
    runGoldenMaster(async () => {
        const item1 = new Item("Backstage passes to a TAFKAL80ETC concert", -1, 60);
        const itemsArray = [item1];
        const gr = new GildedRose(itemsArray)
        gr.updateQuality()
    })
})

test("sellIn inférieur à 0, nom != TAFKAL80ETC et quality <50", async () => {
    runGoldenMaster(async () => {
        const item1 = new Item("Backstage passes to a TAFKAL80ETC concert", -1, 40);
        const itemsArray = [item1];
        const gr = new GildedRose(itemsArray)
        gr.updateQuality()
    })
})