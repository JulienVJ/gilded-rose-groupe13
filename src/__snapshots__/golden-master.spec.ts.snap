// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ceci est un test 1`] = `
[
  "31 Quality is less than 50', 'Quality win 1",
  "Aged Brie is Not Sulfuras item so sell In decrease of 1 ",
]
`;

exports[`Test with quality < 50 1`] = `
[
  "31 Quality is less than 50', 'Quality win 1",
  "Backstage passes to a TAFKAL80ETC concert Is TAFKAL80ETC concert part",
  "sell In under 11",
  "32 Quality is less than 50 Quality win 1 from 11 sell In",
  "Backstage passes to a TAFKAL80ETC concert is Not Sulfuras item so sell In decrease of 1 ",
]
`;

exports[`Test with quality < 50 but sellIn < 6 1`] = `
[
  "31 Quality is less than 50', 'Quality win 1",
  "Aged Brie is Not Sulfuras item so sell In decrease of 1 ",
]
`;
